test/modules/obsagent-client/src/main.rs

```diff
@@ -0,0 +1,210 @@
+/// value-observer).  The rate at which the metrics are updated is configurable. The
+/// rate at which metrics are exported (or pushed) to the OTLP endpoint is also configurable.
+//
+/// The program is configurable via command line arguments as well as environment variables,
+/// with the environment variables taking precendence. The following is a list of the environment
+/// variables (and cmd line args in parenthesis):
+///     1. UPDATE_RATE (--update-rate/-u) - Rate at which each instrument is
+///            updated with a new metric measurement (updates/sec)
+///     2. PUSH_RATE (--push-rate/-p) - Rate at which measurements are pushed
+///            out of the client (pushes/sec)
+///     3. OTLP_ENDPOINT (--otlp-endpoint/-e) - Endpoint to which OTLP messages
+///             will be sent.
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+    init_logging();
+    let config = init_config()?;
+    info!(
+        "Starting Observability Agent Client with configuration: {:?}",
+        config
+    );
+    let _started = init_meter(1.0 / config.push_rate, config.otlp_endpoint)?;
+    let meter = global::meter(METER_NAME);
+
+    // Init synchronous instruments
+    let counter = meter.u64_counter("u64_counter_example").init();
+    let ud_counter = meter
+        .i64_up_down_counter("i64_up_down_counter_example")
+        .init();
+    let value_recorder = meter
+        .f64_value_recorder("f64_value_recorder_example")
+        .init();
+
+    // Init asynchronous instruments
+    let sum = Arc::new(Mutex::new(0));
+    let sum_clone = Arc::clone(&sum);
+    let sum_observer_cb = move |res: ObserverResult<u64>| {
+        if let Ok(sum_clone) = sum_clone.try_lock() {
+            res.observe(*sum_clone, &[]);
+        } else {
+            error!("try_lock failed")
+        }
+    };
+    let _sum_observer = meter
+        .u64_sum_observer("u64_sum_observer_example", sum_observer_cb)
+        .init();
+    let ud_sum = Arc::new(Mutex::new(0));
+    let ud_sum_clone = Arc::clone(&ud_sum);
+    let ud_sum_observer_cb = move |res: ObserverResult<i64>| {
+        if let Ok(ud_sum_clone) = ud_sum_clone.try_lock() {
+            res.observe(*ud_sum_clone, &[]);
+        } else {
+            error!("try_lock failed")
+        }
+    };
+    let _ud_sum_observer = meter
+        .i64_up_down_sum_observer("i64_up_down_sum_observer_example", ud_sum_observer_cb)
+        .init();
+    let value: Arc<Mutex<f64>> = Arc::new(Mutex::new(0.0));
+    let value_clone = Arc::clone(&value);
+    let value_observer_cb = move |res: ObserverResult<f64>| {
+        if let Ok(value_clone) = value_clone.try_lock() {
+            res.observe(*value_clone, &[]);
+        } else {
+            error!("try_lock failed")
+        }
+    };
+    let _value_observer = meter
+        .f64_value_observer("f64_value_observer_example", value_observer_cb)
+        .init();
```

Comment:
How are these observers getting used. Do we have to call the observe method?
https://docs.rs/opentelemetry/0.14.0/opentelemetry/metrics/struct.UpDownSumObserver.html